docs/configuredgraphfactory.txt

```diff
@@ -0,0 +1,440 @@
+about the backend you wish to use when accessing a graph-- every time
+you open a graph.
+
+2.  If your ConfigurationManagementGraph is configured with a distributed
+storage backend then your graph configurations are available to all
+JanusGraph nodes in your cluster.
+
+[[how-does-the-configuredgraphfactory-work]]
+=== How Does the ConfiguredGraphFactory Work?
+
+The `ConfiguredGraphFactory` provides an access point to graphs under
+two scenarios:
+
+1.  You have already created a configuration for your specific graph
+object using the `ConfigurationGraphManagememt#createConfiguration`. In
+this scenario, your graph is opened using the previously created
+configuration for this graph.
+2.  You have already created a template configuration using the
+`ConfigurationManagementGraph#createTemplateConfiguration`. In this
+scenario, we create a configuration for the graph you are creating by
+copying over all attributes stored in your template configuration and
+appending the relevant graphName attribute, and we then open the graph
+according to that specific configuration.
+
+[[accessing-your-graphs]]
+=== Accessing Your Graphs
+
+You can either use `ConfiguredGraphFactory.create("graphName")`--
+or `ConfiguredGraphFactory.open("graphName")`. Learn more about the difference
+between these two options by reading the section below about the `ConfigurationManagementGraph`.
+
+[[listing-your-graphs]]
+=== Listing Your Graphs
+
+`ConfiguredGraphFactory.getGraphNames()` will return a `Set<String>` of graph names
+for which you have created configurations using the `ConfigurationManagementGraph` APIs.
+
+`JanusGraphFactory.getGraphNames()` on the other hand returns a `Set<String>` of graph names
+for which you have instantiated _and_ the references are stored inside the `JanusGraphManager`.
+
+[[configuring-your-server-to-use-the-configuredgraphfactory]]
+=== Configuring Your Server To Use the ConfiguredGraphFactory
+
+To be able to use the `ConfiguredGraphFactory`, you must configure your
+server to use the `ConfigurationManagementGraph` APIs. To do this, you
+have to inject a graph variable named "ConfigurationManagementGraph" in your
+server's YAML's `graphs` map. For example:
+
+[source, properties]
+----
+graphManager: org.janusgraph.graphdb.management.JanusGraphManager
+graphs: {
+  graph: conf/gremlin-server/janusgraph-cassandra-es-server.properties,
+  ConfigurationManagementGraph: conf/janusgraph-configurationmanagement.properties
+}
+----
+
+In this example, our `ConfigurationManagementGraph` graph will be
+configured using the properties stored inside
+`conf/janusgraph-configurationmanagement.properties`, which for
+example, look like:
+
+[source, properties]
+----
+gremlin.graph=org.janusgraph.core.JanusGraphFactory
+storage.backend=cassandrathrift
+graph.graphname=ConfigurationManagementGraph
+storage.hostname=127.0.0.1
+cache.db-cache = true
+cache.db-cache-clean-wait = 20
+cache.db-cache-time = 180000
+cache.db-cache-size = 0.5
```

Comment:
Remove the `cache.*` properties here, even if they remain in the actual properties file, because it is unrelated noise for the example.